src/main/webapp/assets/script.js

```diff
@@ -72,12 +73,89 @@ function onLoad() {
   collegeLocation = fetchFakeCollegeLocation();
 
   // Add the embedded map to the page.
-  const script = document.createElement('script');
-  script.src = `https://maps.googleapis.com/maps/api/js?key=AIzaSyB_iMemNeUtO8D7kGTmagTgluRlsiFTcDA&callback=initMap`;
-  script.defer = true;
-  script.async = true;
-  window.initMap = initMap;
-  document.head.appendChild(script);
+  getSecretFor('javascript-maps-api').then((key) => {
+    const script = document.createElement('script');
+    script.src = `https://maps.googleapis.com/maps/api/js?key=${key}&callback=initMap`;
+    script.defer = true;
+    script.async = true;
+    window.initMap = initMap;
+    document.head.appendChild(script);
+  });
+}
+
+/**
+ * Gets the secret value corresponding to a secret ID from GCP secrets store.
+ * @param {string} secretid - The secret's id, as defined in the secrets store.
+ */
+async function getSecretFor(secretid) {
```

Comment:
I'm wondering if storing the key in a separate JS file might be faster?

Reply:
This solution allows us to store our keys using GCP's secrets management infrastructure, which has several benefits over just using a local file that we .gitignore, mostly due to its centralized nature:

- If we need to regenerate a key because the old one was compromised, we can just update the key in the console, and it will update the key retrieved by all versions of our app, even deployed ones! (no redeploying necessary)
- We don't have to manually figure out where to keep the secrets file and make sure that everyone has their .gitignores configured correctly - if there were a key to update in the .gitignore file, everyone would have to manually update their files, which becomes more of a hassle the more keys we are dealing with
- Also, the work for this version is already done haha

Reply:
Good explanation, Melanie.